conftest.py

```diff
@@ -221,6 +224,37 @@ def pytest_terminal_summary(
         terminalreporter.write("\n" * 2)
 
 
+_MLFLOW_VERSION_NOT_FOUND_ERROR_REGEX = re.compile(
+    r"No matching distribution found for mlflow==([0-9]+\.[0-9]+\.[0-9]+(\.dev[0-9]+)?)"
+)
+
+
+def pytest_exception_interact(node, call, report):
+    """This hook is called when an exception was raised during testing."""
+    exception = call.excinfo.value
+    if not report.failed or not exception:
+        return
+
+    # If it failed to install mlflow dev version, suggest using `--serve-wheel` flag.
+    # This particularly happens during testing, some test cases try to install mlflow
+    # via pip e.g. model loading. They pins mlflow version to install based on current
+    # environment i.e. dev version ahead of the latest release, hence it's not found
+    # on PyPI. `--serve-wheel` flag was introduced to resolve this issue, which starts
+    # local PyPI server and serve a wheel based on local source code.
+    # Ref: https://github.com/mlflow/mlflow/pull/10247
+    m = _MLFLOW_VERSION_NOT_FOUND_ERROR_REGEX.search(str(exception))
+    if m and m.group(1) == VERSION:
+        raise MlflowException(
+            f"The version of MLflow you're attempting to install ({m.group(1)}) is"
+            " not found on PyPI. This could be due to specifying a local development"
+            " version like 2.8.1.dev0, which is one micro version ahead of the latest"
+            " official release. To resolve this issue, you can use the `--serve-wheel`"
+            " flag with your pytest command, for example: `pytest tests --serve-wheel`."
+            " This flag sets up a local PyPI server, allowing you to install MLflow"
+            " from the local source code."
+        )
```

Comment:
Does this work for a test like this?

```python
import subprocess
import sys

import mlflow


def test_foo():
    subprocess.run(
        [sys.executable, "-m", "pip", "install", f"mlflow=={mlflow.__version__}"], check=True
    )

```

Reply:
When `capture_output = True`, yes, otherwise not.